### PR TITLE
Initial service contract docs + autosol

### PR DESCRIPTION
**Pull request by karlb**
Goals of this commit:
* Provide basis for more detailed service contract docs
* Incorporate https://github.com/raiden-network/spec/issues/120
* Incorporate https://github.com/raiden-network/raiden-contracts/issues/447
* Allow automatic generation of docs from solidity docstrings

More details, links, diagrams, etc. will follow in the next PR.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,47 +47,52 @@ help:
 
 .PHONY: clean
 clean:
-	rm -rf $(BUILDDIR)/*
+	rm -rf $(BUILDDIR)/* contracts
+
+# download `raiden-contracts/raiden_contracts/contracts` to the local `contracts` dir
+# This is skipped if the dir already exists. Use `make clean` to force an update.
+contracts:
+	curl -sSL https://github.com/raiden-network/raiden-contracts/tarball/master | tar xvz --wildcards '*/raiden_contracts/contracts' --strip-components 2
```
Comment (palango) [inline Makefile]: I get the following error on macOS, can we workaround that?
```sh
curl -sSL https://github.com/raiden-network/raiden-contracts/tarball/master | tar xvz --wildcards '*/raiden_contracts/contracts' --strip-components 2
tar: Option --wildcards is not supported
```

Comment (palango) [reply Makefile]: Additionally, is this run on readthedocs?

Comment (karlb) [reply Makefile]: Apparently `--wildcards` is a gnu extension and the BSD tar is missing that. I'll find an alternative way of doing that.

Why should `readthedocs` be a problem? I thought it was just a linux system with sphinx installed that runs the different `make` targets.

Comment (karlb) [reply Makefile]: Please try again on MacOS, now.

**Pull request merged**